ietf-voucher-constrained-tree.txt

```diff
@@ -1,15 +1,16 @@
-module: ietf-constrained-voucher
-
-  grouping voucher-constrained-grouping
-    +-- voucher
-       +-- created-on?                      yang:date-and-time
-       +-- expires-on?                      yang:date-and-time
-       +-- assertion                        enumeration
-       +-- serial-number                    string
-       +-- idevid-issuer?                   binary
-       +-- pinned-domain-cert?              binary
-       +-- domain-cert-revocation-checks?   boolean
-       +-- nonce?                           binary
-       +-- last-renewal-date?               yang:date-and-time
-       +-- pinned-domain-pubk?              binary
-       +-- pinned-domain-pubk-sha256?       binary
+﻿module: ietf-constrained-voucher
+
+  grouping voucher-constrained-grouping
+    +-- voucher
+       +-- created-on?                      CBOR text string
+       +-- expires-on?                      CBOR text string

```

Comment:
This is probably a good explanation, but it does not follow the YANG-tree part.
Maybe we should amend as "yang:date-and-time (CBOR text string)"
I guess I'm surprised we aren't mapped to a CBOR time format here.

Reply:
The Voucher was using YANG date-and-time, which is encoded as string. So applying the YANG-to-CBOR mapping: that maps to a CBOR string. Not surprising to me, though it is a nuisance for the parsing in a constrained device. 

Now the question is do we want to stay close to the YANG-to-CBOR translated voucher format (not make any optimizations that deviate from it)?   This includes supporting delta encoding, non-delta encoding, and string format date-and-time.
Or, define in our draft an own encoding that is most easy to implement on constrained devices? (E.g. only support delta encoding for shortest numbers and not support the non-delta encoding of SIDs. )

I thought we decided for the first option.